### ISSUE TITLE
Support for PyCharm/IntelliJ/CLion

### ISSUE THREAD

**Issue by dominick-han** [Rockhopper-Technologies/enlighten#32]
**Describe the bug**
enlighten doesn't seem to support "Terminal" in JetBrain's IDEs

**To Reproduce**
Run demo.py in the "Terminal" tab in PyCharm

**Environment (please complete the following information):**
 - Enlighten Version: 1.6.2
 - OS and version: MacOS 10.15.7
 - Console application: PyCharm 2020.2.3
 - Special Conditions: N/A

**Additional context**
Maybe I'm missing something? A brand new PyCharm install, and using the normal Terminal tab
<img width="1392" alt="Screen Shot 2020-10-08 at 11 43 51 AM" src="https://user-images.githubusercontent.com/11525517/95482234-e266e480-095b-11eb-9909-344d58aace14.png">

**Comment by avylove** [comment]
I'm not sure why this isn't working, but I've been able to replicate it. I'll try to dig in and figure out what's happening.

**Comment by avylove** [comment]
It seems this is a bug in the pycharm terminal's handling of CSR. Here is a script to illustrate the issue.

```
import sys
import blessed

term = blessed.Terminal()

# Set scroll region
sys.stdout.write(term.csr(0, 5))

# Print a line at the top of the terminal
sys.stdout.write(term.move(0, 0))
sys.stdout.write("Top of terminal")

# Print a line near the bottom of the terminal
sys.stdout.write(term.move(term.height - 2, 0))
sys.stdout.write(term.clear_eol + "Bottom of terminal")

# Reset terminal
sys.stdout.write(term.csr(0, term.height - 1))
sys.stdout.write(term.move(term.height, 0))
```

In a standard terminal you'll see a message at the top and bottom of the terminal. In PyCharm, you see


![pycharm csr issue](https://user-images.githubusercontent.com/16090444/95532608-b0727400-0995-11eb-84d2-20e1a9cf4246.png)

**Comment by avylove** [comment]
I created an issue to track this
https://youtrack.jetbrains.com/issue/PY-44864

**Comment by dominick-han** [comment]
Thank you so much!

**Comment by dominick-han** [comment]
Would it (potentially) be an issue with blessed?

**Comment by avylove** [comment]
No, Blessed is just for convenience. On a terminal 25 lines high, this is the equivalent of
```
$ echo -en "\x1b[1;6r\x1b[1;1H\x1b[KTop of terminal\x1b[24;1H\x1b[KBottom of terminal\x1b[1;25r\x1b[26;1H"
```
or in Python
```
height = 25
print(f'\x1b[1;6r\x1b[1;1H\x1b[KTop of terminal\x1b[{height - 1};1H\x1b[KBottom of terminal\x1b[1;{height}r\x1b[{height + 1};1H', end='')
```
